[docs] Improve custom TypeScript functions docs (#27288)
expo/expo: docs/pages/custom-builds/functions.mdx

<Step label="2">

## Compile the function

> Functions must be compiled to a single JavaScript file that can be run without installing any Node Module dependencies. The default `build` script for generated functions uses the [ncc](https://github.com/vercel/ncc) to compile your function into a single file with all its dependencies. If you don't have the `ncc` installed globally on your machine, run `npm install -g @vercel/ncc` to install it. Next, run the build script in the **.eas/build/myFunction** directory:

Reviewer: ```suggestion
Functions must be compiled to a single JavaScript file that can be run without installing any dependencies. The default `build` script for generated functions uses [ncc](https://github.com/vercel/ncc) to compile your function into a single file with all its dependencies. If you don't have the `ncc` installed globally on your machine, run `npm install -g @vercel/ncc` to install it. Next, run the build script in the **.eas/build/myFunction** directory:
```

The `build` script generates **build/index.js**. This file must be uploaded to EAS Build as a part of your project archive, so that the builder can run your function. Ensure that the file is not excluded by a **.gitignore** file or **.easignore** file.

</Step>

<Step label="3">

> ## Connect the function to the custom build config

Reviewer: ```suggestion
## Expose the function to the custom build config
```
```suggestion
## Define the function in the custom build config
```
```suggestion
## Import the function in the custom build config
```

I think these are more… better

> > **Note:** The following example assumes that you have already set up a custom build workflow and configured it in your **eas.json**. If you haven't, see our [“Get started with custom builds” guide](/custom-builds/get-started/#create-a-workflow) first.

Reviewer 2: ```suggestion
> **Note**: The following example assumes that you have already set up a custom build workflow and configured it in your **eas.json**. If not, see [Get started with custom builds](/custom-builds/get-started/#create-a-workflow) before proceeding.
```

- No quotes in when referring internal link. The link color highlight is enough to grab attention of a user if they don't skim the note.
- Also, we don't have to explicitly call out the type of the document here.

Author: thanks!

> > **Note:** The following example assumes that you have already set up a custom build workflow and connected it with your **eas.json**. If you haven't, see our ["Get started with custom builds" guide](/custom-builds/get-started/#create-a-workflow) first.

Reviewer: ```suggestion
> **Note:** The following example assumes that you have already set up a custom build workflow and configured it in your **eas.json**. If you haven't, see our [“Get started with custom builds” guide](/custom-builds/get-started/#create-a-workflow) first.
```
